papaparse.js

```diff
@@ -1089,8 +1100,12 @@ License: MIT
 
 		this.resume = function()
 		{
-			_paused = false;
-			self.streamer.parseChunk(_input, true);
+			if(self.streamer.halted()) {
+				_paused = false;
+				self.streamer.parseChunk(_input, true);
+			} else {
+				setTimeout(() => this.resume(), 3);
```

Comment:
Why setting a timeout?

Reply:
To wait for the current process to get to this condition :-
`this._handle.paused() || this._handle.aborted()`
And halt. 3 is the magic number I came up with after brute force (which gives the minimum parse time)

Reply:
Then we should wait on a loop until this condition is true. Using a timeout is a bad idea as it may depend on the performance capabilities of the running hardware.

Reply:
Being a single threaded language, wouldn't we be blocking the said code to actually run if we keep on waiting in a loop?

Tried :- 
```
while(true) {
	if(self.streamer.halted()) {
		_paused = false;
		self.streamer.parseChunk(_input, true);
		return;
	}
}
```
Browser locked up.

Reply:
You are right, then we should call do something like: 

```
setTimeout(function)() {
  this.resume() // or a new function to test the condition and execute the parseChunk
}.bind(this), 10)
```

Reply:
I wouldn't want to create a new function upon every resume. This is working just fine with scope intact :-
`setTimeout(this.resume, 3);`

So if that's settled then I'll implement this halting process for every streamer and add the test case.

Reply:
Great, that makes sense. 

Could you please add a comment explaing why we use the setTimeout? For me it's a big warning on the library an we should always remember why it was added